sympy/integrals/tests/test_deltafunctions.py

```diff
@@ -35,8 +35,9 @@ def test_deltaintegrate():
     assert deltaintegrate(DiracDelta(x)**2, x) == DiracDelta(0)*Heaviside(x)
     assert deltaintegrate(y*DiracDelta(x)**2, x) == \
         y*DiracDelta(0)*Heaviside(x)
-    assert deltaintegrate(DiracDelta(x, 1)**2, x) is None
-    assert deltaintegrate(y*DiracDelta(x, 1)**2, x) is None
+    assert deltaintegrate(DiracDelta(x, 1)**2, x) == -DiracDelta(0, 2)*Heaviside(x)
```

Comment:
I don't know the motivation for the practical motivation for the squared-delta-derivative terms, but internally, they test the delta-function `simplify()` routine.

`deltaintegrate` appears once in `integrals.py`'s function `Integral._eval_integral()`.  It is called for both definite and  indefinite forms.  It seems irregular to me.  `manualintegrate` should be able to handle the cases I have in mind relatively easily, so I'm not sure why it gets special treatment.  It appears that `deltaintegrate` used to be called directly in some past cases (perhaps for speed reasons?) but that seems scarce now (see `test_continuous_rv.py`).

I can add the tests for the un-squared delta-terms.